scripts/docs-csat.py

```diff
@@ -0,0 +1,318 @@
+                    ],
+                    'metrics': [
+                        {'expression': 'ga:totalEvents'}
+                    ],
+                    'dimensions': [
+                        {'name': 'ga:pageTitle'},
+                        {'name': 'ga:pagePath'}
+                        # {'name': 'ga:eventAction'}
+                    ],
+                    'dimensionFilterClauses': [
+                        {
+                            'filters': [
+                                {
+                                    'dimensionName': 'ga:eventCategory',
+                                    'operator': 'REGEXP',
+                                    'expressions': ['docs-feedback-yes']
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+    ).execute()
+
+
+def get_pageviews(analytics):
+    return analytics.reports().batchGet(
+        body= {
+            'reportRequests': [
+                {
+                    'viewId': args.ga_view_id,
+                    'dateRanges': [
+                        {'startDate': args.start_date, 'endDate': args.end_date}
+                    ],
+                    'metrics': [
+                        {'expression': 'ga:uniquePageviews'},
+                    ],
+                    'dimensions': [
+                        {'name': 'ga:pageTitle'},
+                        {'name': 'ga:pagePath'}
+                    ],
+                    'dimensionFilterClauses': [
+                        {
+                            'filters': [
+                                {
+                                    'dimensionName': 'ga:pagePath',
+                                    'operator': 'IN_LIST',
+                                    'expressions': [pages]
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+    ).execute()
+
+
+no_votes = dict()
+yes_votes = dict()
+pageviews = dict()
+def create_dict(response, dict):
+    for report in response.get('reports', []):
+
+        for row in report.get('data', {}).get('rows', []):
+            page = row.get('dimensions')[1]
+            # title = row.get('dimensions')[0]
+            # if not title.isascii():
+            #     continue
+            metrics = row.get('metrics')
+
+            for i, values in enumerate(metrics):
+                value = values.get('values')[0]
+                if page in dict:
+                    dict[page] += float(value)
+                    continue
+                dict[page] = float(value)
+
+                pages.append(page)
+
+    # print(dict)
+
+
+average_csats = dict()
+weighted_csats = dict()
+def calculate_csat(no_votes, yes_votes, pageviews):
+
+    # Remove blacklisted pages.
+    for p in blacklist:
```

Comment:
Should have blacklist passed as a parameter, rather than using the global blacklist.